File: alphatimer.api.protocol/src/main/java/de/dennisfabri/alphatimer/api/protocol/events/messages/DataHandlingMessage2.java
package de.dennisfabri.alphatimer.api.protocol.events.messages;

import de.dennisfabri.alphatimer.api.protocol.events.messages.enums.TimeInfo;
import de.dennisfabri.alphatimer.api.protocol.events.messages.enums.TimeMarker;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@AllArgsConstructor
@EqualsAndHashCode
@ToString
@Getter
public class DataHandlingMessage2 implements Message {
    private final byte lane;
    private final byte currentLap;
    private final int timeInMillis;
    private final TimeInfo timeInfo;
    private final TimeMarker timeMarker;
}

File: protocol-api/src/main/java/de/dennisfabri/alphatimer/api/protocol/events/messages/DataHandlingMessage2.kt
package de.dennisfabri.alphatimer.api.protocol.events.messages

import de.dennisfabri.alphatimer.api.protocol.events.messages.enums.TimeInfo
import de.dennisfabri.alphatimer.api.protocol.events.messages.enums.TimeMarker

data class DataHandlingMessage2(
    val original: String,
    val lane: Byte,
    val currentLap: Byte,
    val timeInMillis: Int,
    val timeInfo: TimeInfo,
    val timeMarker: TimeMarker
) : Message
